Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -1429,14 +1430,15 @@ - (void)_runningOrTerminatingServiceWorkerCountForTesting:(void(^)(NSUInteger))c
     });
 }
 
-- (void)_fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(WK_SWIFT_UI_ACTOR void (^)(NSData *))completionHandler
+- (void)_fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(void(^)(NSData *))completionHandler
 {
     if ([dataTypes containsObject:WKWebsiteDataTypeLocalStorage]) {
         _websiteDataStore->fetchLocalStorage([completionHandler = makeBlockPtr(completionHandler)](HashMap<WebCore::ClientOrigin, HashMap<String, String>>&& localStorage) {
             constexpr unsigned currentLocalStorageSerializationVersion = 1;
 
             WTF::Persistence::Encoder encoder;
             encoder << currentLocalStorageSerializationVersion;
+            encoder << HashSet<String> { WKWebsiteDataTypeLocalStorage };
```

Comment:
I don't think we should encode a String since this is very inefficient. Ideally, this would be a enum.

Also, we don't think we want to encode a HashSet. A Vector would be better so we have strict ordering.

Reply:
Updated to use enum. I've asked a question about the rest on Slack.